### PR TITLE
Order charging bl

### PR DESCRIPTION
**Pull request by RivkaEhrenfeld**
find orders by multple status2

### CODE REVIEW

**Review by avigailRo — COMMENTED** [review]
כל הכבוד!

File: src/main/java/com/sap/ordermanagergreen/controller/OrderController.java [hunk 1]
```diff
@@ -1,39 +1,60 @@
...
 import com.sap.ordermanagergreen.util.JwtToken;
+import lombok.SneakyThrows;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+
+
+
```
Comment (geniuscodes) [inline src/main/java/com/sap/ordermanagergreen/controller/OrderController.java:17]: spacing

File: src/main/java/com/sap/ordermanagergreen/controller/OrderController.java [hunk 2]
```diff
@@ -1,39 +1,60 @@
...
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-import static com.sap.ordermanagergreen.OrderManagerGreenApplication.MY_URL;
-
-@CrossOrigin(MY_URL)
 @RestController
+@CrossOrigin("http://localhost:3000")
```
Comment (geniuscodes) [inline src/main/java/com/sap/ordermanagergreen/controller/OrderController.java:26]: put this somewhere as constant

File: src/main/java/com/sap/ordermanagergreen/repository/IOrderRepository.java [hunk 3]
```diff
@@ -10,5 +10,12 @@
...
 public interface IOrderRepository extends MongoRepository<Order, String> {
-    List<Order> findByOrderStatusAndCompany_Id(Pageable pageable, OrderStatus orderStatusId, String companyName);
+List<Order>findByOrderStatusInAndCompanyId(Pageable pageable, List<OrderStatus> orderStatus,String companyId);
+
+
+
+
+//List<Order> findByOrderStatusAndCompanyId(Pageable pageable, OrderStatus orderStatus,String companyId);
+
+
```
Comment (geniuscodes) [inline src/main/java/com/sap/ordermanagergreen/repository/IOrderRepository.java:20]: spacing

File: src/main/java/com/sap/ordermanagergreen/service/OrderChargingService.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                } else {
+                    p.setInventory(p.getInventory() - item.getQuantity());
+                    productRepository.save(p);
+                }
+            }
+
+            OrderDTO orderDTO = this.modelMapper.map(order, OrderDTO.class);
+            //push orderDto to queue
+
+
```
Comment (geniuscodes) [inline src/main/java/com/sap/ordermanagergreen/service/OrderChargingService.java:37]: spacing

File: src/main/java/com/sap/ordermanagergreen/service/ProductService.java [hunk 5]
```diff
@@ -42,7 +42,11 @@ public List<Product> get(String token) {
         return modelMapper.map(products, listType);
     }
 
-    public Map<String, String> getNames(String token, String prefix) {
+@SneakyThrows
+    public Map<String,String> getNames(String token, String prefix) {
+
+
+
```
Comment (geniuscodes) [inline src/main/java/com/sap/ordermanagergreen/service/ProductService.java:50]: spacing

File: src/main/java/com/sap/ordermanagergreen/service/ProductService.java [hunk 6]
```diff
@@ -74,7 +83,11 @@ public void update(String id, Product product, String token) throws ObjectExistE
         productRepository.save(product);
     }
 
-    public void delete(String id, String token) throws NoPermissionException {
+@SneakyThrows
+    public void delete(String id, String token)throws NoPremissionException {
+
+
+
```
Comment (geniuscodes) [inline src/main/java/com/sap/ordermanagergreen/service/ProductService.java:91]: spacing

File: src/main/java/com/sap/ordermanagergreen/util/JwtToken.java [hunk 7]
```diff
@@ -29,15 +28,19 @@ public static String generateToken(User user) {
         return accessToken;
     }
 
-    public static TokenDTO decodeToken(String token) {
+    public static TokenDTO decodeToken(String token) throws TokenNotValidException {
+        try{
+
+
+
```
Comment (geniuscodes) [inline src/main/java/com/sap/ordermanagergreen/util/JwtToken.java]: spacing

**Review by geniuscodes — CHANGES_REQUESTED** [review]

**Pull request closed without merging**